### PR TITLE
feat(console): support managing user mfa on user details page

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Support managing user mfa on user details page

### Update
- Extract `MfaFactorTitle` and `MfaFactorName` into components
- Refactor `MfaForm` swith field with `MfaFactorTitle`

### Note
- The UI tests for this feature is not available since it depends on mfa verification binding

### Todo
- LOG-7097: Add UI tests for managing user mfa

## Testing

Test locally.

image

image

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/components/MfaFactorName/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export type Props = {
+  type: MfaFactor;
+  agent?: string;
+};
+
+function MfaFactorName({ type, agent }: Props) {
+  return (
+    <>
+      <DynamicT forKey={factorNameLabel[type]} />
+      {agent && ` - ${agent}`}
```
Comment (wangsijie) [inline packages/console/src/components/MfaFactorName/index.tsx:21]: The "agent" is "User-Agent" which need to parse and format before display, would you mind creating an issue for this?

File: packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    data: mfaVerifications,
+    error,
+    isLoading,
+    mutate,
+  } = useSWR<UserMfaVerificationResponse, RequestError>(`api/users/${userId}/mfa-verifications`);
+
+  const api = useApi();
+  const { show: showConfirm } = useConfirmModal();
+
+  const handleDelete = async (mfaVerification: UserMfaVerification) => {
```
Comment (simeng-li) [inline packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx:31]: nit: useCallback?

File: packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      return;
+    }
+
+    await api.delete(`api/users/${userId}/mfa-verifications/${mfaVerification.id}`);
+    void mutate(mfaVerifications?.filter((item) => item.id !== mfaVerification.id));
+  };
+
+  return (
+    <>
+      {!isLoading && !error && (
```
Comment (simeng-li) [inline packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx:55]: will this cause page jump while missing elements?

File: packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx]: Updated.

File: packages/console/src/components/MfaFactorName/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/components/MfaFactorName/index.tsx:21]: Created.
LOG-7104

File: packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx:59]: No, `useSWR` will not trigger loading state when revalidating existing data.

**Review by simeng-li — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]

**Pull request merged**